### ISSUE TITLE
Missing run button for tests with the same title

### ISSUE THREAD

**Issue by alexdima** [microsoft/vscode#144755]
In our sources, create two tests e.g.:
```ts
	test('a', () => {
	});
	test('a', () => {
	});
```
The tests run fine and this doesn't appear to be forbidden by mocha:
<img width="463" alt="image" src="https://user-images.githubusercontent.com/5047891/157447622-cf436350-280a-44fc-8922-b9eb4060337d.png">

However, in the editor, only the second test has a run button:
<img width="258" alt="image" src="https://user-images.githubusercontent.com/5047891/157447676-f83bcbc0-452a-4453-acf9-72a03ba5de24.png">

cc @jrieken @hediet

**Comment by connor4312** [comment]
This is because they look identical to the test extension, so the last one wins. When Mocha reports test results, even to the raw 'reporter' interface, it does so only by using the completed title of the test, so in the results I could not distinguish between two identically named tests.

**Comment by alexdima** [comment]
@connor4312 I created the issue because I ran into this 2-3 times in the past. I think I might be using a pattern where I duplicate a test in order to write a new one. The missing run icon made me think that testing is not flaky and made me go back to the command line. Only yesterday I realized the root cause of the problem is the title duplication. I agree that having two tests with the same title is unreasonable and doesn't need to be supported. But would it be possible to show a squiggle, or render the buttons and then run both tests, or some other solution? It's just the current solution made me think testing is flaky, because there's no indication that the title is duplicated, especially with tests which have many lines and don't fit in a viewport.
